docs/user/autoyast.md

```diff
@@ -348,17 +346,84 @@ section.
-implementation point of view because we might need to extract AutoYaST scripts and put them in some
-place for Agama to run them.
+AutoYaST implements five kind of scripts: `pre`, `post-partitioning`, `chroot`, `post`, and `init`.
+In Agama there are only three of them, although the original `pre-scripts` behavior is preserved
+when importing an AutoYaST profile.
+
+The following table summarizes the equivalences:
+
+| AutoYaST          | When they run                   | Agama equivalent                       |
+| ----------------- | ------------------------------- | -------------------------------------- |
+| pre               | before the installation starts  | pre, see [Pre-scripts](#pre-scripts)   |
+| post-partitioning | after the partitioning is done  | none                                   |
+| chroot            | after the installation finishes | post                                   |
+| post              | before the 2nd stage            | none, as there is no 2nd stage         |
+| init              | at the end of the 1st boot      | init                                   |
+
+The pre-scripts are processed by AutoYaST code itself, and the `post` and `init` scripts are merged
+into a single type (`init`) which runs during the 1st boot after the installation.
+
+#### Scripts representation
+
+A definition of an AutoYaST script supports several features. However, in Agama we are implementing
+only those we consider essential.
+
+| AutoYaST      | Supported | Agama  | Comment               |
+| ------------- | --------- | ------ | --------------------- |
+| file_name     | Yes       | name   |                       |
+| location      | Yes       | url    |                       |
+| source        | Yes       | body   |                       |
+| rerun         | No        |        |                       |
+| interpreter   | Partial   |        | Script's shebang      |
+| feedback      | No        |        |                       |
+| feedback_type | No        |        |                       |
+| debug         | No        |        |                       |
+| notification  | No        |        |                       |
+| param-list    | Planned   |        |                       |
+| chrooted      | Yes       | chroot | Only for post-scripts |
+
+#### pre-scripts
+
+AutoYaST pre-scripts are executed before the installation and are a powerful method to customize the
+profile at runtime. If you are using an AutoYaST profile, `pre-scripts` will work in the same way
+with Agama. Please, check the [Dynamic profiles](#dynamic-profiles) section for further details.
+
+However, Agama ships its own pre-scripts mechanism, although they do not allow modifying the profile
+itself because using [Jsonnet](https://jsonnet.org/) is the preferred way to do that. The main use
+cases for Agama pre-scripts are activating hardware, tweaking the installation media, etc.
```

Comment:
so pre scripts are always run before probe, right?

Reply:
Well, that's a tricky question. In a single product it might not be the case. But instead of forcing a probing, I think we need decide if it makes sense to have separate workflows for both cases.

Reply:
At this point we do not ship any image with a single product, so it is not *that urgent*, but something to tackle sooner than later.